libraries/AP_HAL_SITL/sitl_gps.cpp

```diff
@@ -1082,24 +1081,37 @@ uint32_t SITL_State::CalculateBlockCRC32(uint32_t length, uint8_t *buffer, uint3
 /*
   temporary method to use file as GPS data
  */
-void SITL_State::_update_gps_file(const struct gps_data *d)
+void SITL_State::_update_gps_file(uint8_t instance)
 {
     static int fd = -1;
-    if (fd == -1) {
-        fd = open("/tmp/gps.dat", O_RDONLY|O_CLOEXEC);
+    static int fd2 = -1;
+    int temp_fd;
+    if (instance == 0) {
+        temp_fd = fd;
+        if (temp_fd == -1) {
+            temp_fd = open("/tmp/gps.dat", O_RDONLY|O_CLOEXEC);
+        }
+        fd = temp_fd;
```

Comment:
Can we do this simpler? It took me a bit to understand the logic here.

```
if (instance == 0) {
    if (fd == -1) {
        fd = open("/tmp/gps.dat", O_RDONLY|O_CLOEXEC);
    }
    temp_fd = fd;
```

Same below.